### PR TITLE
dev: lint ADR phase prompts for the delegation-contract blocks

### PR DESCRIPTION
**Pull request by andrebrait**
Follow-up to the delegation contract (7ce4573b) and companion to #921/#922: the mechanical floor for spec quality. An `/adr-phase` implementer executes a phase prompt verbatim, so a prompt missing the contract blocks ships weak work before any gate can catch it.

## What

- **`scripts/check_phase_prompts.py`** — flags a post-contract phase prompt (ADR ≥ 60; older ADRs grandfathered, in file-argument mode too) missing any of: a `VERIFICATION` section, a `HANDOFF` block naming its `RESULTS/` file, a declared test mode (`red-run`/`red->green` vs `behaviour-preserving`/`oracle`), or the reality-override line. Presence only — depth stays the orchestrator gate's job.
- **Pre-commit**: runs when `.ADRs/**.txt` is staged. **CI**: `test_repo_tree_is_clean` in the pytest suite gates every tracked post-contract prompt.
- **`/spec-lint`** skill wraps it; **`/adr-create`** now runs it before reporting a new ADR done.
- Drive-by: capitalised a `# shellcheck (…)` comment in `.githooks/pre-commit` that ShellCheck parsed as a malformed directive (SC1073), unblocking `shellcheck` on the hook itself.

## Evidence

- Red→green: tests written first, run against the absent module → collection error; after implementation → `14 passed`.
- Live hook runs: staged non-compliant `ADR_60` prompt → pre-commit `rc=1` naming all four missing blocks; compliant prompt → `rc=0`.
- Gates: `python -m pytest` → 2245 passed, 4 skipped; `ruff check`/`ruff format --check` clean; `mypy tests/` clean; `sh -n` + `shellcheck --severity=info` on the hook clean bar the pre-existing SC2016 info (deliberate literal `$var` in a step label, untouched).

## Coverage matrix

Every check red+green paired; each mode token parametrised; grandfather cutoff (ADR_59 non-compliant → clean); RESULTS handoffs and non-numeric `.txt` skipped; fully non-compliant prompt reports all four; repo-tree scan green.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01PfJUsCQkpFZxXufncGUS2f

## Summary by CodeRabbit

* **New Features**
  * Added automatic validation for ADR phase prompts to catch missing required sections and formatting issues before completion.
  * Introduced a new prompt-checking workflow that reports clear pass/fail results and flags unreadable files.

* **Bug Fixes**
  * Strengthened pre-commit checks so ADR-related prompt files are validated consistently, helping prevent incomplete or noncompliant prompts from being submitted.

### CODE REVIEW

File: scripts/check_phase_prompts.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+
+def _check_prompt(text: str) -> list[tuple[str, str]]:
+    """Return ``(check_id, message)`` for each contract block missing in ``text``."""
+    lower = text.lower()
+    missing: list[tuple[str, str]] = []
+    if not re.search(r"^VERIFICATION", text, flags=re.MULTILINE):
+        missing.append(("verification-section", "no VERIFICATION section"))
+    if not (re.search(r"^HANDOFF", text, flags=re.MULTILINE) and "RESULTS/" in text):
+        missing.append(("handoff-results", "no HANDOFF block naming its RESULTS/ file"))
```
[the remark this answers is not in the document]
Comment (andrebrait) [reply scripts/check_phase_prompts.py]: Applied in 53f346ab — the parallel Sonnet adversarial review reproduced the same exploit (a REQUIRED READING citation of the prior phase's `RESULTS/` file satisfying the check while the HANDOFF block named nothing). `_handoff_names_results()` now requires the `RESULTS/` mention inside the HANDOFF span (the `HANDOFF` line and its continuation up to the next all-caps section header), pinned red→green by `test_handoff_results_must_be_in_the_handoff_span`.

---
🤖 Generated by [Claude Code](https://claude.com/claude-code), posted via @andrebrait's account on their behalf.

File: scripts/check_phase_prompts.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    are skipped."""
+    violations: list[tuple[Path, str, str]] = []
+    for path in paths:
+        adr = _prompt_adr_number(path)
+        if adr is None or adr < _CONTRACT_MIN_ADR:
+            continue
+        try:
+            text = path.read_text(encoding="utf-8", errors="replace")
+        except OSError:
+            continue
```
[the remark this answers is not in the document]
Comment (andrebrait) [reply scripts/check_phase_prompts.py:139]: Applied in 53f346ab — an unreadable gated prompt now surfaces as its own `unreadable` violation (exit 1) instead of silently counting as compliant, pinned red→green by `test_unreadable_prompt_is_a_loud_violation`. (The parallel Sonnet adversarial review flagged the same fail-open pattern.)

---
🤖 Generated by [Claude Code](https://claude.com/claude-code), posted via @andrebrait's account on their behalf.

**Comment by andrebrait** [comment]
## Review audit — PR #927

**Reviews received & shape.** Sonnet 5 adversarial review, single-agent at effort `xhigh` (size metric: 5 files / ~330 lines dev tooling, below the multi-agent threshold), execution-grounded (~18 hostile fixtures run). A second Sonnet `xhigh` **delta re-review** covered the review-fix commits. **GitHub Copilot** reviewed (2 findings). **CodeRabbit: QUOTA** — "Review limit reached" with no review content for the whole window → did-not-review, not a clean pass. **Snyk: QUOTA** — `code/snyk` status `error` ("Code test limit reached") → scan skipped, not a security pass.

**Findings ledger (every finding, one outcome each):**

| # | Source | Finding | Outcome |
| - | ------ | ------- | ------- |
| 1 | Sonnet (blocking) | `handoff-results` satisfied by a `RESULTS/` mention anywhere (e.g. REQUIRED READING) | fixed@53f346ab (`_handoff_names_results` span check; red→green test) |
| 2 | Sonnet (nitpick) | mode tokens substring-matched ("coracle" ⇒ oracle) | fixed@53f346ab (word-bounded `_MODE_RE`; red→green test) |
| 3 | Sonnet (nitpick) | negated override line ("do NOT let … override this prompt") passes | skipped: checker is a documented presence-only floor (docstring: depth stays the orchestrator gate's job); reviewer rated low-priority; negation heuristics risk false positives on legitimate nearby phrasing |
| 4 | Sonnet (nitpick) | `git ls-files` cwd-dependent — pytest from a subdir silently defeats the CI gate | fixed@53f346ab (`git -C` repo root; red→green test) |
| 5 | Sonnet (nitpick) | unreadable prompt silently counts as compliant | fixed@53f346ab (loud `unreadable` violation; red→green test) |
| 6 | Sonnet (nitpick) | test module name diverges from sibling spelling | fixed@53f346ab (renamed `test_phase_prompts_check.py`) |
| 7 | Sonnet (outside-diff) | pre-existing SC2016 info surfaced by the SC1073 fix | skipped: pre-existing, info severity, deliberate literal `$var` in a single-quoted step label (untouched line, `git blame` origin/devel) |
| 8 | Copilot | same as #1 | fixed@53f346ab; replied on thread |
| 9 | Copilot | same as #5 | fixed@53f346ab; replied on thread |
| 10 | Delta re-review (blocking) | `_RETROFITTED_ADRS` under-inclusive — ADR-52 (Proposed, 4 non-compliant prompts) missed because a negative status grep ate "Not yet implemented" | fixed@7791347d: ADR-52 prompts retrofitted on devel (9b21b75e); set now pins ALL Proposed ADRs incl. prompt-less 51/56/57; parametrized red→green (4 failed → green) |
| 11 | Delta re-review (nitpick) | violation output printed absolute paths after the cwd fix | fixed@7791347d (repo-relative display, absolute kept for out-of-tree args) |

Delta re-review per-area verdicts: handoff-span edge cases PASS (8 probes), mode-token lookarounds PASS (11 probes), repo-root resolution PASS, rename PASS (no dangling refs), test honesty PASS (all four regression tests executed red against the pre-fix module, green after).

Both external bots being quota-dead was escalated per flow: the second independent Sonnet pass (delta re-review) ran instead of merging on the single review.

---
🤖 Generated by [Claude Code](https://claude.com/claude-code), posted via @andrebrait's account on their behalf.
